### PR TITLE
Add more restriction criteria to prizes

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: frontend/app/src/components/home/PrizeContent.svelte [hunk 1]
```diff
@@ -33,19 +46,31 @@
...
+        content.diamondOnly ||
+        content.lifetimeDiamondOnly ||
+        content.uniquePersonOnly ||
+        content.streakOnly > 0;
+
+    $: userEligible =
+        (!content.diamondOnly || $isDiamond) &&
+        (!content.lifetimeDiamondOnly || $isLifetimeDiamond) &&
+        (!content.uniquePersonOnly || $user.isUniquePerson) &&
+        content.streakOnly <= client.getStreak($user.userId);
```
Comment (julianjelfs) [inline frontend/app/src/components/home/PrizeContent.svelte:68]: You might be better off using `$chitStateStore.streak` here as what you have will not be reactive. Not _particularly_ important in this case, but it _could_ give you the wrong result.

File: frontend/app/src/components/home/PrizeContent.svelte [hunk 2]
```diff
@@ -85,7 +125,43 @@
             </div>
         {/if}
         {#if !me}
-            <div class="click"><Translatable resourceKey={i18nKey("prizes.click")} /></div>
+            {#if restrictedPrize}
+                <div class="restricted">
+                    <Translatable resourceKey={i18nKey("prizes.restrictedMessage")} />
+                    {#if content.diamondOnly || content.lifetimeDiamondOnly}
+                        <div on:click={onDiamondClick}>
```
Comment (julianjelfs) [inline frontend/app/src/components/home/PrizeContent.svelte:132]: Do these things _appear_ clickable?

File: frontend/app/src/components/home/PrizeContentBuilder.svelte [hunk 3]
```diff
@@ -34,12 +34,15 @@
...
     } from "openchat-client";
+    import Checkbox from "../Checkbox.svelte";
+    import Select from "../Select.svelte";
 
     const ONE_HOUR = 1000 * 60 * 60;
     const ONE_DAY = ONE_HOUR * 24;
     const ONE_WEEK = ONE_DAY * 7;
     const client = getContext<OpenChat>("client");
     const dispatch = createEventDispatcher();
+    const streaks = ["3", "7", "14", "30", "100"];
```
Comment (julianjelfs) [inline frontend/app/src/components/home/PrizeContentBuilder.svelte:45]: We _do_ actually have a 365 streak implemented in most places as well.

File: frontend/app/src/components/home/PrizeContent.svelte (hunk 2 above)
Comment (megrogan) [reply frontend/app/src/components/home/PrizeContent.svelte:132]: I use cursor: pointer

File: frontend/app/src/components/home/PrizeContentBuilder.svelte (hunk 3 above)
Comment (megrogan) [reply frontend/app/src/components/home/PrizeContentBuilder.svelte:45]: Well nobody has this yet so perhaps we shouldn't yet allow it as a prize gate

**Review by julianjelfs — DISMISSED** [review]

**Review by julianjelfs — APPROVED** [review]

**Comment by megrogan** [comment]
This can be merged when communities and groups have been upgraded

File: tsBindings/types.d.ts [hunk 4]
```diff
@@ -105,7 +105,7 @@ export type SwapStatusErrorExpired = { token0_txn_out?: bigint | undefined, };
...
-export type ChatMetrics = { text_messages: bigint, image_messages: bigint, video_messages: bigint, audio_messages: bigint, file_messages: bigint, polls: bigint, poll_votes: bigint, icp_messages: bigint, sns1_messages: bigint, ckbtc_messages: bigint, chat_messages: bigint, kinic_messages: bigint, deleted_messages: bigint, giphy_messages: bigint, prize_messages: bigint, prize_winner_messages: bigint, replies: bigint, edits: bigint, reactions: bigint, proposals: bigint, reported_messages: bigint, message_reminders: bigint, custom_type_messages: bigint, last_active: bigint, };
+export type ChatMetrics = { text_messages: bigint, image_messages: bigint, video_messages: bigint, audio_messages: bigint, file_messages: bigint, polls: bigint, poll_votes: bigint, crypto_messages: bigint, icp_messages: bigint, sns1_messages: bigint, ckbtc_messages: bigint, chat_messages: bigint, kinic_messages: bigint, deleted_messages: bigint, giphy_messages: bigint, prize_messages: bigint, prize_winner_messages: bigint, replies: bigint, edits: bigint, reactions: bigint, proposals: bigint, reported_messages: bigint, message_reminders: bigint, custom_type_messages: bigint, last_active: bigint, };
```
Comment (hpeebles) [inline tsBindings/types.d.ts:108]: This `crypto_messages` field can't go out on the frontend until groups and communities are upgraded

Comment (megrogan) [reply tsBindings/types.d.ts:108]: Notice my comment - this PR is already waiting on groups and communities...

Comment (hpeebles) [reply tsBindings/types.d.ts:109]: Also can't go out until Users are upgraded, since they return metrics too

**Review by hpeebles — APPROVED** [review]

**Pull request merged**